### PR TITLE
Add tag support for redshift module

### PR DESCRIPTION
**Pull request by rafaeldriutti**
##### SUMMARY
Follow up after https://github.com/ansible/ansible/pull/56981
Adds tag support to reshift module
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

redshift.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:21]: With the migration to collections, the 'old' versions have been removed.
```suggestion
```

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:97]:
```suggestion
```

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:105]:
```suggestion
```

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:181]:
```suggestion
    version_added: "1.2.0"
```

And for brand new features this needs to use the next collection version.

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:187]:
```suggestion
    version_added: "1.2.0"
```

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:190]:
```suggestion
  - amazon.aws.aws
```

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:191]:
```suggestion
  - amazon.aws.ec2
```

File: plugins/modules/redshift.py
Comment (tremble) [inline plugins/modules/redshift.py:294]: This makes an assumption that we're in the `aws` partition.  This doesn't hold true for China or GovCloud.

I've opened https://github.com/ansible-collections/amazon.aws/pull/136 which adds get_account_info which also returns the partition (using similar logic to get_account_id)

File: tests/integration/targets/redshift/tasks/main.yml
Comment (tremble) [inline tests/integration/targets/redshift/tasks/main.yml:123]: I'd recommend having at least 1 tag using uppercase:
```suggestion
          Tizio: Caio
```

A number of modules use CamelCasing tricks which sometimes result in tag casing to get messed up.

**Review by tremble — APPROVED** [review]
@jillr - since I added a couple of the changes here could you have a quick look at this one please.

File: plugins/modules/redshift.py
Comment (tremble) [reply plugins/modules/redshift.py:291]: Rather than holding this up indefinitely I think we come back and clean this up when we have the helper available to us.

**Review by jillr — APPROVED** [review]
The CI failure should be fixed by #271, we can merge this once CI is green
Thanks @rafaeldriutti

**Comment by tremble** [comment]
CI's green.  Merging.

**Pull request merged**
